stdlib/public/core/CocoaArray.swift

```diff
@@ -83,12 +83,13 @@ internal struct _CocoaArrayWrapper: RandomAccessCollection {
     let result = _ContiguousArrayBuffer<AnyObject>(
       _uninitializedCount: boundsCount,
       minimumCapacity: 0)
-
-    // Tell Cocoa to copy the objects into our storage
-    core.getObjects(
-      UnsafeMutableRawPointer(result.firstElementAddress)
-      .assumingMemoryBound(to: AnyObject.self),
-      range: _SwiftNSRange(location: bounds.lowerBound, length: boundsCount))
+    
+    let base = UnsafeMutableRawPointer(result.firstElementAddress)
+      .assumingMemoryBound(to: AnyObject.self)
+      
+    for idx in 0..<boundsCount {
+      (base + idx).initialize(to: core.objectAt(idx + bounds.lowerBound))
+    }
```

Comment:
Oh, IIUC, the issue is just that `-[NSArray getObjects:range:]` gives us items at a +0 refcount. We should continue to use it -- we just need to explicitly retain them. 

The internal type `_UnmanagedAnyObjectArray` helps with getting refcounts right for the implementation side of `getObjects:`-type ObjC methods -- we should just use it here, too. (We can easily add a method on it that retains the entire contents of the buffer. We ought to add a `count` to it, anyway, for proper bounds checking.)